File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/httprecipes/spider/SpiderException.java
package com.heatonresearch.httprecipes.spider;

/**
 * The Heaton Research Spider 
 * Copyright 2007 by Heaton Research, Inc.
 * 
 * HTTP Programming Recipes for Java ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 * 
 * SpiderException: This exception is thrown when the spider
 * encounters an error.
 * 
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 * 
 * @author Jeff Heaton
 * @version 1.1
 */
public class SpiderException extends Exception {

	/**
	 * Serial id for this class.
	 */
  private static final long serialVersionUID = 1L;

  /**
   * Construct a SpiderException.
   * 
   * @param message
   *          The message for the exception.
   */
  public SpiderException(String message) {
    super(message);
  }

  /**
   * Wrap another exception as a SpiderException.
   * 
   * @param t
   *          The exception that occured.
   */
  public SpiderException(Throwable t) {
    super(t);
  }
}

File: CSIntroNeuralNetworkEdition2/HeatonResearchSpider/Spider/SpiderException.cs
// The Heaton Research Spider for .Net 
// Copyright 2007 by Heaton Research, Inc.
// 
// From the book:
// 
// HTTP Recipes for C# Bots, ISBN: 0-9773206-7-7
// http://www.heatonresearch.com/articles/series/20/
// 
// This class is released under the:
// GNU Lesser General Public License (LGPL)
// http://www.gnu.org/copyleft/lesser.html
//
using System;
using System.Collections.Generic;
using System.Text;

namespace HeatonResearch.Spider
{
    /// <summary>
    /// SpiderException: This exception is thrown when the spider
    /// encounters an error.
    /// </summary>
    public class SpiderException:Exception
    {
        /// <summary>
        /// Throw a spider exception.
        /// </summary>
        /// <param name="msg">The exception message.</param>
        public SpiderException(String msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Pass on an exception as a SpiderException.
        /// </summary>
        /// <param name="e">The other exception.</param>
        public SpiderException(Exception e)
            : base("Exception while spidering", e)
        {
        }
    }
}
